buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -129,14 +129,16 @@ class NodeInfo {
             'JAVA_HOME' : project.javaHome,
             'ES_GC_OPTS': config.jvmArgs // we pass these with the undocumented gc opts so the argline can set gc, etc
         ]
-        args.add("-Des.node.portsfile=true")
-        args.addAll(config.systemProperties.collect { key, value -> "-D${key}=${value}" })
+        args.addAll("-E", "es.node.portsfile=true")
+        args.addAll(config.esSettings.collectMany { key, value -> ["-E", "${key}=${value}" ] })
+        env.put('ES_JAVA_OPTS', config.systemProperties.collect { key, value -> "-D${key}=${value}" }.join(" "))
```

Comment:
`JAVA_OPTS` is not officially supported, it looks like a few projects use it (most notably Tomcat). The [officially-supported environment variable](https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/envvars002.html) for this purpose is `JAVA_TOOL_OPTIONS`, but [some issues on Ubuntu break the usage](https://github.com/elastic/elasticsearch/issues/13785) of that so we do not support it. We publicly document `ES_JAVA_OPTS`, and do not publicly document `ES_JAVA_OPTS`. Since we can not use the standard one, do not document `JAVA_OPTS` and it's not really official, I see no reason to break the documented `ES_JAVA_OPTS` here and think that we should just get rid of supporting `JAVA_OPTS`.